Drv/Ip/SocketReadTask.cpp

```diff
@@ -51,34 +51,42 @@ SocketIpStatus SocketReadTask::open() {
 }
 
 void SocketReadTask::shutdown() {
+    this->m_task_lock.lock();
```

Comment:
@JohanBertrand @csmith608 These likely need to be discarded as they affect locking.